### ISSUE TITLE
report - Unable to process POST

### ISSUE THREAD

**Issue by TristisOris** [zammad/zammad#1846]
### Infos:

* Used Zammad version:  2.3.0-1519143336.d8f56634.xenial amd64
* Installation method (source, package, ..): package
* Operating system: ubuntu server 16.04.3
* Database + version:  mysql-client-core-5.7 ; mariadb-client-core-10.0
* Elasticsearch version:
* Browser + version: ff58

### Expected behavior:

*

### Actual behavior:
when open page /#report
got a modal window:
```

StatusCode: 500
{"error":"Unable to process POST at http://127.0.0.1:9200/zammad_production/Ticket/_search\n#\u003cUserAgent::Result:0x00007fddd85181b0 @success=false, @body=nil, @data=nil, @code=0, @content_type=nil, @error=\"#\u003cErrno::ECONNREFUSED: Failed to open TCP connection to 127.0.0.1:9200 (Connection refused - connect(2) for \\\"127.0.0.1\\\" port 9200)\u003e\"\u003e"}
```

### Steps to reproduce the behavior:
open page /#report
and any actions on this page.

it's not a production installation, i'm just testing this product. So i don't making any configurations yet.
only install from official guide.

**Comment by thorsteneckel** [comment]
Your elasticsearch is not running. Make sure it's up and running on the printed address.

We should catch this error and print it out in a more user friendly way because there are a lot of issues with the same question because of the very technical error message here.

**Comment by TristisOris** [comment]
because it was not installed( sorry sorry sorry.
i installed elastic, but trouble still exist.

`{"error":"Unable to process POST at http://localhost:9200/zammad_production/Ticket/_search\n#\u003cUserAgent::Result:0x00007f2af03013e0 @success=false, @body=nil, @data=nil, @code=\"400\", @content_type=nil, @error=\"Client Error: #\u003cNet::HTTPBadRequest 400 Bad Request readbody=true\u003e!\"\u003e"}`

but i can explain, why too many questions about it ^_^
i didn't see what it must be istalled, before you say about elastic search.
![default](https://user-images.githubusercontent.com/1270117/36467949-4f99dd14-16f2-11e8-838c-b9d84f4f9d0e.png)

This link are totaly invisible, because:
1. it not in prerequisites.
2. you already start checking locale.

just edit the guide a little, please =)

**Comment by sheLbyPL** [comment]
Check if your elasticsearch process is running - `systemctl status elasticsearch`

If yes, check that you have ingest-attachment installed:
`sudo /usr/share/elasticsearch/bin/elasticsearch-plugin list`
It should return "ingest-attachment"

If you have it, then run this command:
`zammad run rake searchindex:rebuild`

**Comment by TristisOris** [comment]
yes, of course.
` elasticsearch.service - Elasticsearch
   Loaded: loaded (/usr/lib/systemd/system/elasticsearch.service; enabled; vendor preset: enabled)
   Active: active (running) since Wed 2018-02-21 10:35:09 MSK; 9min ago
     Docs: http://www.elastic.co
  Process: 1009 ExecStartPre=/usr/share/elasticsearch/bin/elasticsearch-systemd-pre-exec (code=exited, status=0/SUCCE
 Main PID: 1027 (java)
    Tasks: 44
   Memory: 1.3G
      CPU: 19.455s
   CGroup: /system.slice/elasticsearch.service
           └─1027 /usr/bin/java -Xms2g -Xmx2g -XX:+UseConcMarkSweepGC -XX:CMSInitiatingOccupancyFraction=75 -XX:+UseC
`
nope.
`sudo /usr/share/elasticsearch/bin/elasticsearch-plugin list
mapper-attachments
`

i already made rebuild a twice and restarted VM.

**Comment by thorsteneckel** [comment]
@TristisOris - you can create a pull request for the documentation [over here](https://github.com/zammad/zammad-documentation) and help us and the community.

**Comment by thorsteneckel** [comment]
@TristisOris - please check your elasticsearch version. Looks like you're using version 6 but only 5.5 is supported currently.

**Comment by TristisOris** [comment]
curl -XGET 'localhost:9200'
```
{
  "name" : "ZMZaLMA",
  "cluster_name" : "elasticsearch",
  "cluster_uuid" : "GeU5h1MTS6ue6JRF4pxvhg",
  "version" : {
    "number" : "5.6.8",
    "build_hash" : "688ecce",
    "build_date" : "2018-02-16T16:46:30.010Z",
    "build_snapshot" : false,
    "lucene_version" : "6.6.1"
  },
  "tagline" : "You Know, for Search"
}
```

**Comment by thorsteneckel** [comment]
What's going on in your elasticsearch log?

**Comment by sheLbyPL** [comment]
I think wrong elasticsearch plugin is the problem.
Do this:
`sudo /usr/share/elasticsearch/bin/elasticsearch-plugin remove mapper-attachments`

Then:
`sudo /usr/share/elasticsearch/bin/elasticsearch-plugin install ingest-attachment`

Then:
`zammad run rake searchindex:rebuild`

**Comment by TristisOris** [comment]
but instruction says to install mapper-attachments. https://docs.zammad.org/en/latest/install-elasticsearch.html#ubuntu-16-04

but looks like it help.

i'm repeated all installation process, but it was too long, so i increased VM from 2cpu/2gb to 4cpu/4gb.
Just for information, because i'm not only reinstalled this plugin.

All works now, thank you. i must to read instructions carefully(

**Comment by monotek** [comment]
Which elasticsearch version do you use?

**Comment by TristisOris** [comment]
as in install guide.

curl -XGET 'localhost:9200'
```
{
  "name" : "ZMZaLMA",
  "cluster_name" : "elasticsearch",
  "cluster_uuid" : "GeU5h1MTS6ue6JRF4pxvhg",
  "version" : {
    "number" : "5.6.8",
    "build_hash" : "688ecce",
    "build_date" : "2018-02-16T16:46:30.010Z",
    "build_snapshot" : false,
    "lucene_version" : "6.6.1"
  },
  "tagline" : "You Know, for Search"
}
```

**Comment by monotek** [comment]
@TristisOris 
> but instruction says to install mapper-attachments

That was wrong. 

If you read the whole page its clear that mapper-ingest-attachment is for Elasticsearch 2.4 - 5.5. 
From Elasticsearch 5.6 you need to use ingest-attachment.

I've changed the docs...

**Comment by TristisOris** [comment]
ok, thank you.

**Comment by thorsteneckel** [comment]
Reopening since wen need to make the shown error message more userfriendly to avoid issues like this.

**Comment by thorsteneckel** [comment]
We improved the elastic search error handling to get [humanized error messages](https://github.com/zammad/zammad/commit/8c5f87c9d4118849bc2783db59997322b4064318) and do a [version check for incompatible versions](https://github.com/zammad/zammad/commit/c479fba6646c19d1d1f503398d94cdca4baca6ad). This will be available in Zammad 2.4. Closing now.
